pkg/services/validationService.go

```diff
@@ -114,95 +114,92 @@ func (vs *ValidationService) Validate(admissionReviewReq *admission.AdmissionRev
-		if err == nil {
-			cliEvaluationId = evaluationResultResp.EvaluationId
-		} else {
-			cliEvaluationId = -2
-			internalLogger.LogAndReportUnexpectedError("saving evaluation results failed")
-			*warningMessages = append(*warningMessages, "saving evaluation results failed")
+		// evaluate policy against configuration
+		policyCheckResults, err := evaluator.Evaluate(policyCheckData)
+		if err != nil {
+			internalLogger.LogAndReportUnexpectedError(fmt.Sprintf("Evaluate err: %s", err.Error()))
 		}
-	}
 
-	resultStr, err := evaluation.GetResultsText(&evaluation.PrintResultsData{
-		Results:           results,
-		EvaluationSummary: evaluationSummary,
-		LoginURL:          prerunData.RegistrationURL,
-		Printer:           printer.CreateNewPrinter(),
-		K8sVersion:        clusterK8sVersion,
-		Verbose:           os.Getenv(enums.Verbose) == "true",
-		PolicyName:        policy.Name,
-		OutputFormat:      os.Getenv(enums.Output),
-	})
+		results := policyCheckResults.FormattedResults
+		passedPolicyCheckCount := 0
+		if results.EvaluationResults != nil {
+			passedPolicyCheckCount = results.EvaluationResults.Summary.FilesPassedCount
+		}
 
-	if err != nil {
-		internalLogger.LogAndReportUnexpectedError(fmt.Sprintf("GetResultsText err: %s", err.Error()))
-	}
+		evaluationSummary := getEvaluationSummary(policyCheckResults, passedPolicyCheckCount)
+
+		// send results to backend
+		noRecords := os.Getenv(enums.NoRecord)
+		if noRecords != "true" {
+			evaluationResultResp, err := vs.sendEvaluationResult(vs.getEvaluationRequestData(policy.Name, startTime,
+				policyCheckResults, namespace, resourceKind, resourceName))
+			if err == nil {
+				cliEvaluationId = evaluationResultResp.EvaluationId
+			} else {
+				cliEvaluationId = -2
+				internalLogger.LogAndReportUnexpectedError("saving evaluation results failed")
+				*warningMessages = append(*warningMessages, "saving evaluation results failed")
+			}
+		}
 
-	var allowed bool
-	var isFailedPolicyCheck = evaluationSummary.PassedPolicyCheckCount == 0
-	if isFailedPolicyCheck {
-		allowed = false
+		// get results text
+		resultStr, err := evaluation.GetResultsText(&evaluation.PrintResultsData{
+			Results:           results,
+			EvaluationSummary: evaluationSummary,
+			LoginURL:          prerunData.RegistrationURL,
+			Printer:           printer.CreateNewPrinter(),
+			K8sVersion:        clusterK8sVersion,
+			Verbose:           os.Getenv(enums.Verbose) == "true",
+			PolicyName:        policy.Name,
+			OutputFormat:      os.Getenv(enums.Output),
+		})
+		if err != nil {
+			internalLogger.LogAndReportUnexpectedError(fmt.Sprintf("GetResultsText err: %s", err.Error()))
+		}
 
-		sb := strings.Builder{}
-		sb.WriteString("\n---\n")
-		sb.WriteString(resultStr)
-		msg = sb.String()
-	} else {
-		allowed = true
+		didFailCurrentPolicyCheck := evaluationSummary.PassedPolicyCheckCount == 0
+		if didFailCurrentPolicyCheck {
+			didFailAtLeastOnePolicyCheck = true
```

Comment:
think about remove `didFailAtLeastOnePolicyCheck` and use  `allowed`

Reply:
we decided not to, because of line 202